domain/tutoringModel/nestor.py

```diff
@@ -0,0 +1,281 @@
+            "{'VAM': 'Yes'}": "zusatzmaterial_visuell",
+            "{'TAM': 'Yes'}": "zusatzmaterial_textuell",
+        }
+        # this dict maps FSLSM model learning style nomenclature
+        # used in HASKI to nomenclature
+        # used in training Nestor.
+        self.ls_map_common_HASKI_to_nestor = {
+            "act": "Active",
+            "ref": "Reflective",
+            "sns": "Sensory",
+            "int": "Intuitive",
+            "vis": "Visual",
+            "vrb": "Verbal",
+            "seq": "Sequential",
+            "glo": "Global",
+        }
+        # self.random_seed = random.seed(143)
+
+    def train_nestor(self):
+        """This function is used to train the BN and save weights locally
+        The saved weights are further used by the get_learning_path method
+        to return the learning path
+        """
+        print(
+            "The scripts used for training is stored in a seperate folder"
+            "along with helper functions and configuration variables"
+        )
+        return None
+
+    def get_learning_path(
+        self,
+        input_learning_style: dict,
+        input_learning_elements: list,
+        path_to_model=os.path.join(
+            "domain", "tutoringModel", "NestorFolder", "backup_saved_model.xml"
+        ),
+    ):
+        """
+        This function reads in pre-processed information
+        of learner
+        to recommend personalized learning paths
+        """
+        random.seed(ran_seed)
+        # defining the path to saved BN
+        # path_to_model = path_to_model
+        # The HASKI Gemeninsam uses the Learning styles naming convention of
+        # (act, ref, sen, int, vis, vrb, seq, glo).
+        # The Nesor uses a naming convention of
+        # (Active, Reflective, Sensory, Intuitive,
+        # Visual, Verbal, Sequential, Global)
+        # Below a dict is used to map this naming convention for giving
+        # the evidence on student to nestor
+        # before, first it is checked if the input learning style is
+        # empty or no to raise errors
+        if not input_learning_style:
+            raise err.MissingParameterError()
+        else:
+            evidence_for_inference = {
+                "Active_Reflective_Dim": self.ls_map_common_HASKI_to_nestor[
+                    input_learning_style["processing_dimension"]
+                ],
+                "Sensory_Intuitive_Dim": self.ls_map_common_HASKI_to_nestor[
+                    input_learning_style["perception_dimension"]
+                ],
+                "Visual_Verbal_Dim": self.ls_map_common_HASKI_to_nestor[
+                    input_learning_style["input_dimension"]
+                ],
+                "Sequential_Gloabl_Dim": self.ls_map_common_HASKI_to_nestor[
+                    input_learning_style["understanding_dimension"]
+                ]
+                # no bfi and listk are used
+                # "bfie": str(bfie_val),
```

Comment:
is this a comment or commented out?

Reply:
Removed!!

It is just a comment. The BFI and LIST-k responses can also be used as evidence to Nestor's inference, but it has no effect on learning paths (for now).